bundles/AdminBundle/Controller/Admin/AssetController.php

```diff
@@ -2268,6 +2268,10 @@ public function gridProxyAction(Request $request, EventDispatcherInterface $even
                 $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() == '/' ? "'/%'" : $list->quote($folder->getRealFullPath() . '/%'));
             }
 
+            if ($allParams['only_unreferenced'] == 'true') {
+                $conditionFilters[] = 'id NOT IN (SELECT targetid FROM dependencies WHERE targettype=\'asset\')';
```

Comment:
I'm afraid that this could cause major performance issues on bigger installations, I guess we should try to reduce the result set in the sub-select as much as possible.

Reply:
I do not think that this is a problem as MySQL uses indexes for the subquery:
```SQL
explain select * FROM `assets` WHERE (path LIKE '/%' AND id NOT IN (SELECT targetid FROM dependencies WHERE targettype='asset') AND type != 'folder') ORDER BY `id` ASC LIMIT 25
```
gives me:
```
id	select_type	table	partitions	type	possible_keys	key	key_len	ref	rows	filtered	Extra
1	PRIMARY	assets		index	"fullpath,path"	PRIMARY	4		50	45.00	Using where
2	DEPENDENT SUBQUERY	dependencies		index_subquery	"targetid,targettype"	targetid	4	func	13	31.35	"Using where; Using index"
```
So there is no temporary table or anything worse than this.
For 30.000 assets this query is executed on our database (with SQL_NO_CACHE to get the performance without query cache) in under 0.5 seconds.